login-fire-button.html

```diff
@@ -195,16 +195,17 @@
     },
 
     /**
-     * The provider logo must be hidden if the element can sign out and a user
-     * is currently signed in.
+     * The provider logo is shown when the user can sign in and is not signed in
+     * Or when they can not sign out. It is hidden otherwise.
      *
+     * @param  {Boolean} noSignIn   is `false` if a user can sign in
      * @param  {Boolean} noSignOut is `false` if the element can sign out
      * @param  {Boolean} signedIn   is `true` if a user is actually signed in
      * @return {Boolean}            `true` if the element can sign out and a
      * user is currently signed in.
      */
     _computeHideIcon: function(noSignIn, noSignOut, signedIn) {
-      return noSignIn || (!noSignOut && signedIn);
+      return !((!noSignIn && !signedIn) || noSignOut);
```

Comment:
I would describe it like this:

_"The button should not display social provider icon when the user is not yet signed in or it can't be used to login out. So, the icon has to be hidden in those two cases"_

IMHO, my version is easier to read but the logic staying the same is still a little complicated to understand. 😕 . Keep the one you prefer 😉 .